File: src/main/java/fi/aalto/cs/apluscourses/utils/parser/O1NewsParser.java
package fi.aalto.cs.apluscourses.utils.parser;

import org.jetbrains.annotations.NotNull;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

public class O1NewsParser implements NewsParser {
  @NotNull
  private final String language;

  public O1NewsParser(@NotNull String language) {
    this.language = language;
  }

  @Override
  public @NotNull String parseTitle(@NotNull Element titleElement) {
    return getElementsByLanguage(titleElement).first().text();
  }

  @Override
  public @NotNull String[] parseBody(@NotNull Element bodyElement) {
    return getElementsByLanguage(bodyElement).stream().map(Element::text).toArray(String[]::new);
  }

  private Elements getElementsByLanguage(@NotNull Element element) {
    var elements = element.getElementsByClass("only" + language);
    if (elements.isEmpty()) {
      return element.getElementsByClass("onlyen");
    }
    return elements;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/utils/temp/parser/O1NewsParser.kt
package fi.aalto.cs.apluscourses.utils.temp.parser

import org.jetbrains.annotations.NonNls
import org.jsoup.nodes.Element
import org.jsoup.select.Elements

class O1NewsParser(private val language: String) : NewsParser() {

    override fun parseTitle(titleElement: Element): String {
        return getElementsByLanguage(titleElement).first().text()
    }

    override fun parseBody(bodyElement: Element): String {
        return getElementsByLanguage(bodyElement).html()
    }

    private fun getElementsByLanguage(element: Element): Elements {
        @NonNls val className = "only$language"
        @NonNls val defaultClassName = "onlyen"
        val elements = element.getElementsByClass(className)
        if (elements.isEmpty) {
            return element.getElementsByClass(defaultClassName)
        }
        return elements
    }

    companion object {
        const val NAME: String = "O1"
    }
}
